bin/node-template/node/src/service.rs

```diff
@@ -42,7 +44,7 @@ macro_rules! new_full_start {
 					.ok_or_else(|| sc_service::Error::SelectChainRequired)?;
 
 				let (grandpa_block_import, grandpa_link) =
-					grandpa::block_import(client.clone(), &*client, select_chain)?;
+					grandpa::block_import(client.clone(), &(client.clone() as Arc<_>), select_chain)?;
```

Comment:
```suggestion
					grandpa::block_import(client.clone(), &(client.clone() as _), select_chain)?;
```

Have you tried this?

Reply:
yup, here's the error you get.

```
  ::: bin/node-template/node/src/command.rs:35:32
   |
35 |                 |config: _| Ok(new_full_start!(config).0),
   |                                ----------------------- in this macro invocation

error[E0620]: cast to unsized type: `alloc::sync::Arc<sc_client::client::Client<sc_client_db::Backend<sp_runtime::generic::block::Block<sp_runtime::generic::header::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, sc_client::call_executor::LocalCallExecutor<sc_client_db::Backend<sp_runtime::generic::block::Block<sp_runtime::generic::header::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>, sc_executor::native_executor::NativeExecutor<service::Executor>>, sp_runtime::generic::block::Block<sp_runtime::generic::header::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>, node_template_runtime::RuntimeApi>>` as `dyn sc_finality_grandpa::GenesisAuthoritySetProvider<sp_runtime::generic::block::Block<sp_runtime::generic::header::Header<u32, sp_runtime::traits::BlakeTwo256>, sp_runtime::OpaqueExtrinsic>>`
  --> bin/node-template/node/src/service.rs:47:45
   |
47 |                     grandpa::block_import(client.clone(), &(client.clone() as _), select_chain)?;
   |                                                            ^^^^^^^^^^^^^^^^^^^^^
...
```